File: svws-webclient/core/src/core/utils/uv/UvKursImportQuellkurs.ts
import { JavaObject } from '../../../java/lang/JavaObject';
import { ArrayList } from '../../../java/util/ArrayList';
import type { List } from '../../../java/util/List';
import { Class } from '../../../java/lang/Class';

export class UvKursImportQuellkurs extends JavaObject {

	/**
	 * Die ID des Quellkurses.
	 */
	readonly id: number;

	/**
	 * Die ID des Fachs.
	 */
	readonly idFach: number;

	/**
	 * Die Kursart.
	 */
	readonly kursart: string;

	/**
	 * Die Kursnummer.
	 */
	readonly nummer: number;

	/**
	 * Die Wochenstunden.
	 */
	readonly wochenstunden: number;

	/**
	 * Die IDs der zugeordneten Lehrer.
	 */
	readonly lehrerIds: List<number> = new ArrayList<number>();

	/**
	 * Die IDs der zugeordneten Schüler.
	 */
	readonly schuelerIds: List<number> = new ArrayList<number>();

	/**
	 * Die IDs der zugeordneten Schienen.
	 */
	readonly schienenIds: List<number> = new ArrayList<number>();


	/**
	 * Erstellt einen normalisierten Quellkurs.
	 *
	 * @param id die ID des Quellkurses
	 * @param idFach die Fach-ID
	 * @param kursart die Kursart
	 * @param nummer die Kursnummer
	 * @param wochenstunden die Wochenstunden
	 */
	constructor(id: number, idFach: number, kursart: string, nummer: number, wochenstunden: number) {
		super();
		this.id = id;
		this.idFach = idFach;
		this.kursart = kursart;
		this.nummer = nummer;
		this.wochenstunden = wochenstunden;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.core.utils.uv.UvKursImportQuellkurs';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.core.utils.uv.UvKursImportQuellkurs'].includes(name);
	}

	public static readonly class = new Class<UvKursImportQuellkurs>('de.svws_nrw.core.utils.uv.UvKursImportQuellkurs');

}

export function cast_de_svws_nrw_core_utils_uv_UvKursImportQuellkurs(obj: unknown): UvKursImportQuellkurs {
	return obj as UvKursImportQuellkurs;
}

File: svws-core/src/main/java/de/svws_nrw/core/utils/uv/UvKursImportQuellkurs.java
package de.svws_nrw.core.utils.uv;

import java.util.ArrayList;
import java.util.List;

import jakarta.validation.constraints.NotNull;

/** Interne, transpilierbare Normalform eines aus einer Quelle gelesenen Kurses. */
final class UvKursImportQuellkurs {

	/** Die ID des Quellkurses. */
	final long id;
	/** Die ID des Fachs. */
	final long idFach;
	/** Die Kursart. */
	final @NotNull String kursart;
	/** Die Kursnummer. */
	final int nummer;
	/** Die Wochenstunden. */
	final double wochenstunden;
	/** Die IDs der zugeordneten Lehrer. */
	final @NotNull List<Long> lehrerIds = new ArrayList<>();
	/** Die IDs der zugeordneten Schüler. */
	final @NotNull List<Long> schuelerIds = new ArrayList<>();
	/** Die IDs der zugeordneten Schienen. */
	final @NotNull List<Long> schienenIds = new ArrayList<>();

	/**
	 * Erstellt einen normalisierten Quellkurs.
	 *
	 * @param id die ID des Quellkurses
	 * @param idFach die Fach-ID
	 * @param kursart die Kursart
	 * @param nummer die Kursnummer
	 * @param wochenstunden die Wochenstunden
	 */
	UvKursImportQuellkurs(final long id, final long idFach, final @NotNull String kursart, final int nummer, final double wochenstunden) {
		this.id = id;
		this.idFach = idFach;
		this.kursart = kursart;
		this.nummer = nummer;
		this.wochenstunden = wochenstunden;
	}
}
